lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -13,18 +15,22 @@ def self.enabled
 
   def self.reset_priority_by_ordered_ids(ids)
     ids.each_with_index do |id, priority|
-      MiqAeDomain.where(:id => id).first.try(:update_attributes, :priority => priority + 1)
+      MiqAeDomain.find_by!(:id => id).update_attributes(:priority => priority + 1)
```

Comment:
@gmcculloug 
This code is used by the UI once it has asked us for a list of domains under the tenant. The UI does the sort within that group and sends us back the ID's in the new priority order.